src/platform/DeviceInfoProvider.cpp

```diff
@@ -82,11 +82,16 @@ CHIP_ERROR DeviceInfoProvider::AppendUserLabel(EndpointId endpoint, const UserLa
 {
     size_t length;
 
-    // Increase the size of UserLabelList by 1
+    // Fetch current list length
     ReturnErrorOnFailure(GetUserLabelLength(endpoint, length));
-    ReturnErrorOnFailure(SetUserLabelLength(endpoint, length + 1));
 
-    // Append the user label at the end of UserLabelList
+    if (length >= kMaxUserLabelListLength)
+    {
+        return CHIP_ERROR_NO_MEMORY;
+    }
```

Comment:
- Please add a unit test
- Please cover removal using the test

Reply:
added new test to cover my modification, and removal has already been covered in https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/TestUserLabelCluster.yaml#L55